electrum/address_synchronizer.py

```diff
@@ -200,11 +200,11 @@ def start_network(self, network: Optional['Network']) -> None:
             self.asyncio_loop = network.asyncio_loop
             self.register_callbacks()
 
+    @with_lock
     @event_listener
     def on_event_blockchain_updated(self, *args):
-        with self.lock:
-            self._get_balance_cache = {}  # invalidate cache
-            self.db.put('stored_height', self.get_local_height())
+        self._get_balance_cache = {}  # invalidate cache
+        self.db.put('stored_height', self.get_local_height())
```

Comment:
This technically accidentally works, but it would make much more sense to chain the function decorators in the other order:

```py
    @event_listener
    @with_lock
    def on_event_blockchain_updated(self, *args):
```
The only reason it works as-is, is because `event_listener` refers to functions by `name`.

However, the order I recommend fails for another reason, that I now fixed in https://github.com/spesmilo/electrum/commit/4887fb3e7ff8d842648b8477903bc5ce24941894, so you would need to rebase.

Reply:
Yes, I initially wrote it in that order (just spuriously) and it exploded, so I changed it. Rebased + applied.